lib/commands/skill/add-locales/helper.js

```diff
@@ -0,0 +1,130 @@
+    const iModelSourceByLocales = _resolveSourcesForIModel(selectedLocales, profile, doDebug);
+    const uniqueTemplateSet = new Set(iModelSourceByLocales.values());
+    const iModelFolderPath = path.join(process.cwd(), ResourcesConfig.getInstance().getSkillMetaSrc(profile),
+        CONSTANTS.FILE_PATH.SKILL_PACKAGE.INTERACTION_MODEL, 'custom');
+    // copy iModel JSONs that contains same language
+    iModelSourceByLocales.forEach((v, k) => {
+        if (fs.existsSync(v)) {
+            const sourceLocale = path.basename(v, path.extname(v));
+            const targetFilePath = path.join(iModelFolderPath, `${k}.json`);
+            fs.copySync(v, targetFilePath); // do not fail if the target locale exists already
+            uniqueTemplateSet.delete(v);
+            const sourceManifestLocale = Manifest.getInstance().getPublishingLocale(sourceLocale);
+            Manifest.getInstance().setPublishingLocale(k, sourceManifestLocale);
+        }
+    });
+    // retrieve remote templates when no existing JSON shares the same language
+    _retrieveTemplatesByLanguage(uniqueTemplateSet, doDebug, (templateErr, templateByLanguage) => {
+        if (templateErr) {
+            return callback(templateErr);
+        }
+        iModelSourceByLocales.forEach((v, k) => {
+            if (!fs.existsSync(v)) {
+                const targetFilePath = path.join(iModelFolderPath, `${k}.json`);
+                fs.writeFileSync(targetFilePath, templateByLanguage.get(v), { encoding: 'utf-8' });
+                Manifest.getInstance().setPublishingLocale(k, { name: 'please change' });
+            }
+        });
+        Manifest.getInstance().write();
+        callback(undefined, iModelSourceByLocales);
+    });
+}
+
+/**
+ * Based on user's selected locales, and the local project's iModel files, decide where to get the
+ * iModel file to add in user's local project. Exclude reusableLocales themselves from the result.
+ *
+ * @param {Array} selectedLocales
+ * @param {String} profile
+ * @param {Boolean} doDebug
+ * @returns {Map} { locale: filePath/remoteURI }
+ */
+function _resolveSourcesForIModel(selectedLocales, profile, doDebug) {
+    const skillMetaController = new SkillMetadataController({ profile, doDebug });
+    const localIModelByLocale = skillMetaController.getInteractionModelLocales();
+    const result = new Map();
+    for (const locale of selectedLocales) {
+        if (R.keys(localIModelByLocale).includes(locale)) {
+            continue;
+        }
+        const languageExtractor = (s) => R.slice(0, R.lastIndexOf('-', s), s);
+        const language = languageExtractor(locale);
+        const reusableLocale = R.find((k) => languageExtractor(k) === language)(R.keys(localIModelByLocale));
+        if (reusableLocale) {
+            result.set(locale, localIModelByLocale[reusableLocale]);
+        } else {
+            result.set(locale, CONSTANTS.TEMPLATES.INTERACTION_MODEL_BY_LANGUAGE[language]);
```

Comment:
would it makes sense store the templates locally? seem like a risky assumption to rely on skill-sample-nodejs-hello-world repo. I dont think being dynamic provide much value here unless we have a better source of truth.

Reply:
what about choosing the same approach as our skill template way? It is hosting a map in s3 with { language: templateUrl }, so that it serves as a dynamic link to the template, so that we can update the map any time without backward compatibility issue.

Reply:
1) Do expect it to be updated often? I think we have mix of both. for example default template for CF deployer is stored in the source code.
2) it introduces extra dependency. we need to make sure upload to s3 is automated. For example, we updated manually, but forget to set public permission - this would be a possible risk if not automated.
3) If it needs to be updated, since it is on Github it would be easier to for contributer just to update template on Github and open a PR.

Both ways have pros and cons. I would feel better about storing on S3 if we had automated S3 upload.

Reply:
Template should be up to the template owner to decide, usually it's not frequent since hello world logic is static and simplified. Also, we don't upload the file, but just store the linkage between language to the template url in the s3, so that template won't be linked with CLI code version.